processing/src/main/java/org/apache/druid/query/ChainedExecutionQueryRunner.java

```diff
@@ -141,34 +142,38 @@ public ChainedExecutionQueryRunner(
                           );
                         }
                     )
-                )
-            );
+                );
 
-            queryWatcher.registerQueryFuture(query, futures);
+            Function<Throwable, Void> cancelFunction = (t) -> {
```

Comment:
I've been thinking about this, and it seems good to use `CompletableFuture` rather than having another home-grown util method for `ListenableFuture`. But it needs more work like how to support the query priority with `CompletableFuture`, it would be fine to add a util method for now.

Reply:
> This little block is used in a few places, and it would be good to include comments about why it's needed.
> 
> Could you please move it to GuavaUtils and also make the following changes:
> 
> * Add a comment about why it's needed: it's an alternative to `Futures.allAsList(...).cancel`, that is necessary because `Futures.allAsList` creates a Future that cannot be canceled if one of its constituent futures has already failed. (This comment is the real reason that it's good to have it be its own function.)
> * The function isn't doing anything with the Throwable, so it could just be `void cancelAll(List<Future<T>>)`.
> * Add unit tests to GuavaUtilsTest.
> 
> Then we can do stuff like `GuavaUtils.cancelAll(futures)`.

Thanks for your advice, I've revised it.